lib/models/repository-states/present.js

```diff
@@ -233,17 +234,27 @@ export default class Present extends State {
         Keys.branches,
       ],
       // eslint-disable-next-line no-shadow
-      () => this.executePipelineAction('COMMIT', (message, options) => {
-        const opts = (!options || !options.coAuthors)
-          ? options
-          : {
-            ...options,
-            coAuthors: options.coAuthors.map(author => {
-              return {email: author.getEmail(), name: author.getFullName()};
-            }),
-          };
+      () => this.executePipelineAction('COMMIT', async (message, options) => {
+        // check with Ash that I retained the logic he was going for...
+        const coAuthors = options && options.coAuthors;
+        const opts = !coAuthors ? options : {
+          ...options,
+          coAuthors: coAuthors.map(author => {
+            return {email: author.getEmail(), name: author.getFullName()};
+          }),
+        };
 
-        return this.git().commit(message, opts);
+        await this.git().commit(message, opts);
+        const {unstagedFiles, mergeConflictFiles} = await this.getStatusesForChangedFiles();
+        const unstagedCount = Object.keys({...unstagedFiles, ...mergeConflictFiles}).length;
+        addEvent('commit', {
+          package: 'github',
+          partial: unstagedCount > 0,
```

Comment:
Since there are several UI options for partial staging, it would be interesting to instrument those as well.  Not something you have to implement here, just something to consider for future iterations.

Reply:
Yeah, I had that thought but then a couple of considerations caused me to pause. Maybe you have thoughts that would help with un-pausing :). 

Do you have specific questions you'd like answered by this information or examples of conclusions we might be able to draw?

These were the considerations that caused me to pause initially:
* as mentioned in the description, tracking all of the staging/unstaging events seems like it could be a lot of noise with little additional value. For example, keyboard users might stage files by hitting `enter` a bunch of times for each line/hunk/file they want to stage... so for a given partial commit, there could be a ton of staging events. Maybe this could provide valuable information, but nothing came to mind for me immediately
* the distinction between partial staging as opposed to staging everything can't necessarily be inferred from instrumenting the UI controls. For example, clicking the "Stage selection" button could result in partial staging of a file, or it could result in staging all changes if the user has selected all of the changes before clicking the button. 

💭 welcome if you have any!